### ISSUE TITLE
Behaviour of return_to query parameter with MFA/2FA

### ISSUE THREAD

**Issue by Benehiko** [ory/kratos#3356]
### Discussed in https://github.com/ory/kratos/discussions/2742

<div type='discussions-op-text'>

<sup>Originally posted by **harnash** September 22, 2022</sup>
Hi.

We have stumbled on an interesting behaviour when using MFA in Kratos with `return_to` parameter.
When login flow is initialized via `/self-service/login/browser` and `return_to` parameter is specified user will be redirected to that URL even when login session is not fully authorized (only first login factor is completed).

Steps to reproduce:
1. Login with user that has MFA enabled but do not enter 2FA code and just return to `/self-service/login/browser` with `return_to` URL set to some arbitrary address
2. Kratos will verify that session is valid but **will not check** if user completed 2FA and will redirect to URL specified in `return_to` parameter causing an infinite loop

Workaround would be to set `refresh` query parameter to 1 but this will force user to re-authenticate.

Is this flow intended?</div>

**Comment by syserr0r** [comment]
I think you need to [specify the `aal` parameter when logging in](https://www.ory.sh/docs/kratos/reference/api#tag/frontend/operation/createBrowserLoginFlow).

The idea being that you can control when the MFA token is required by requiring a higher "authenticator assurance level (AAL)" level.

**Comment by Benehiko** [comment]
@syserr0r Yes you are correct, as mentioned here https://github.com/ory/kratos/discussions/2742#discussioncomment-6325565.

Although this should automatically detect that the user is required an aal2 since the flow right now just continues the `return_to` which seems to be incorrect behavior.

**Comment by syserr0r** [comment]
I believe the reasoning for this is that it would be up to your middleware (the UI portion of Hydra, for which the reference is https://github.com/ory/hydra-login-consent-node) if you wish to have an MFA login or not.

For some users you might only want to do this when they are performing sensitive actions, for others you might only want to do it once for a given time period (i.e. every x weeks) or for every login.

See: https://www.ory.sh/docs/kratos/mfa/overview (last paragraph under 'Overview' above 'Available methods') and the link it references: https://www.ory.sh/docs/kratos/mfa/step-up-authentication

**Comment by mjolk** [comment]
The same happens when you try to recover an account with totp set through a code sent via email. Upon entering the correct code it immediately tries to redirect to the return_to but since it's only partially authenticated it starts bouncing around. When trying ?aal=aal2 i'm getting this error:
```
{
  "id": "cb6481af-c30b-42cf-a6a7-fdae19dd3432",
  "error": {
    "code": 400,
    "status": "Bad Request",
    "reason": "AAL2 login via code requires the `via` query parameter",
    "message": "The request was malformed or contained invalid parameters"
  },
  "created_at": "2024-05-23T20:05:33.812361Z",
  "updated_at": "2024-05-23T20:05:33.812361Z"
}
```
i'm trying to figure out what this error means eg if it's some configuration i'm missing
